### PR TITLE
feat: add Gatsby CLI completion spec

### PR DESCRIPTION
**Pull request by indie-ridge-dev**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

**What is the current behavior? (You can also link to an open issue here)**

**What is the new behavior (if this is a feature change)?**

**Additional info:**

### CODE REVIEW

**Review by cstrnt — CHANGES_REQUESTED** [review]
Hey there :)
Could you please take a look at all `args` here? I just left a comment for some, but I seems that you added `args` for everything, even though they don't take any args!


Here is a little explainer, what an argument is:

`cd /some/folder`

`cd` is the command itself and `/some/folder`is the argument which is passed. Another example from this spec is `gatsby develop -H 3000` where the `-H`is the option and `3000` is the argument from the option

File: dev/gatsby.ts [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+
+        {
+          name: ["staticQueries"],
+          description: "staticQueries",
+          args: {
+            name: "staticQueries",
+          },
+        },
+      ],
+    },
```
Comment (QuiiBz) [inline dev/gatsby.ts:245]: Based on the [documentation](https://www.gatsbyjs.com/docs/reference/gatsby-cli/#repl), I don't think that the REPL takes any options nor arguments. `gatsby repl` will open a REPL, where you can type the commands.

Comment (indie-ridge-dev) [reply dev/gatsby.ts]: yes, you are right of course. I have just pushed a new version without the repl options and without the args for the sub commands.

File: dev/gatsby.ts [hunk 2]
```diff
@@ -0,0 +1,253 @@
+export const completion: Fig.Spec = {
+  name: "gatsby",
+  description: "Gatsby CLI",
+  subcommands: [
+    {
+      name: "develop",
+      description: "Start the Gatsby development server",
+      args: {},
```
[the remark this answers is not in the document]
Comment (QuiiBz) [reply dev/gatsby.ts:8]: Could you remove this argument, as Tim mentioned?

File: dev/gatsby.ts [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        {
+          name: ["-H", "--host"],
+          description: "Set host. Defaults to localhost",
+          args: {},
+        },
+        {
+          name: ["-p", "--port"],
+          description: "Set port. Defaults to env.PORT or 8000",
+          args: {},
+        },
```
Comment (QuiiBz) [inline dev/gatsby.ts:19]: I would love to see `name` filled up in each argument

**Review by QuiiBz — CHANGES_REQUESTED** [review]
Could you remove all `args: {}` where they are not needed, and add a `name` property to each one that really exists?

**Pull request merged**